### DISCUSSION TITLE
Building dynamic ON_CONFLICT  DO_UPDATE

### DISCUSSION THREAD

**Discussion by yuphing-ong** [go-jet/jet#276] [Q&A] [answered]
I have several fields that might be updated, so instead of checking one by one, I tried to do something like:

stmt := table.INSERT(columnList).
MODEL(model).
ON_CONFLICT(table.Col1).DO_UPDATE(
conflictSql
)

and try to build columnList, model and conflictSql dynamically

columnList is of type ColumnList, so I can append table.Col2
model is of type model.<TableName>, so I can assign with model.Col2
but I can't seem to do conflictSql, since its of the form:

SET(table.col2.SET(value))

and SET is a func, so I can't declare var conflictSql []func{} and then append to it with SET(table.Col2.SET(value))

I tried to use an interface instead like var conflictSql []interface{}, and of course I can append, but then I'm not sure how to convert it to what DO_UPDATE expects (seems to be an internal type conflictAction)

**Comment by houtn11** [comment, marked as answer]
I'm not sure what you mean. This should work:

```golang
var columnAssigments []ColumnAssigment

columnAssigments = append(columnAssigments, table.col2.SET(value))
// ...

stmt := table.INSERT(columnList).
	MODEL(model).
	ON_CONFLICT(table.Col1).DO_UPDATE(
		SET(
			columnAssigments...,
		)
	)
```

**Reply by yuphing-ong** [reply]
Ah yes that's exactly what I was looking for, thanks. I didn't append ... to the conflictSQL var
